File: Code/20_Outros/Temporizacao/Ex5.c
#include <stdio.h>
#include <time.h>
#include <unistd.h>

#define N 150
#define DELAY_US 10000 
int main(void)
{
	unsigned int i;
	clock_t CPU_time[2];
	float media;
	CPU_time[0] = clock();
	for(i=1; i<=N; i++)
		usleep(DELAY_US);
	CPU_time[1] = clock();
	media += (float)(CPU_time[1]-CPU_time[0]);
	media /= (float)N;
	media /= (float)CLOCKS_PER_SEC;
	printf("Cálculo de tempo usando clock():\n");
	printf("N = %d\n", N);
	printf("Media de tempo de usleep(%d)\n"
		"   %2.1f us\n",
		DELAY_US, media*1.0e6);
	printf("clock() retorna a quantidade\n");
	printf("   de ciclos de clock consumidos\n");
	printf("   PELO PROCESSO ATUAL.\n");
	printf("A chamada usleep(10000) não é\n");
	printf("   contabilizada.\n");
}

File: 4_Dicas/4.5_Temporização/Python/Ex5.py
import time

N = 150
DELAY_US = 10000
DELAY_S  = DELAY_US*1e-6

CPU_time = [0,0]
CPU_time[0] = time.process_time()
for i in range(N):
	time.sleep(DELAY_S)
CPU_time[1] = time.process_time()
media = (CPU_time[1]-CPU_time[0])/N

print("Cálculo de tempo usando time.process_time():")
print("N = %d" % N)
print("Media de tempo de sleep(%f)\n"
	"   %2.1f us\n" % (DELAY_S, media*1.0e6))
print("time.process_time() retorna a quantidade")
print("   de ciclos de clock consumidos")
print("   PELO PROCESSO ATUAL.")
print("A chamada time.sleep(%f) não é contabilizada.\n" % DELAY_S)

CPU_time[0] = time.thread_time()
for i in range(N):
	time.sleep(DELAY_S)
CPU_time[1] = time.thread_time()
media = (CPU_time[1]-CPU_time[0])/N

print("Cálculo de tempo usando time.thread_time():")
print("N = %d" % N)
print("Media de tempo de sleep(%f)\n"
	"   %2.1f us\n" % (DELAY_S, media*1.0e6))
print("time.thread_time() retorna a quantidade")
print("   de ciclos de clock consumidos")
print("   PELA THREAD ATUAL.")
print("A chamada time.sleep(%f) não é contabilizada." % DELAY_S)